ui-service/src/main/frontend/hotel-booking-ui/src/components/search-components/HotelSearchSectionComponent.tsx

```diff
@@ -0,0 +1,194 @@
+import SearchHotelComponent from "./RoomSelectorComponent.tsx";
+
+
+interface FormData {
+    hotelType: 'single' | 'multi';
+    destination: string;
+    startDate: Date | undefined;
+    endDate: Date | undefined;
+    rooms: number;
+    adults: number;
+    children: number;
+    bundleSave: boolean;
+    addCar: boolean;
+    addFlight: boolean;
+}
+
+function HotelSearchSectionComponent() {
+    const today = new Date();
+    today.setHours(0, 0, 0, 0);
+    const tomorrow = new Date();
+    tomorrow.setDate(today.getDate() + 1);
+    const isDateSelectable = (date: Date) => {
+        return date >= today; // Allow selection of today and future dates
+    };
+
+    const [formData, setFormData] = useState<FormData>({
+        hotelType: 'single',
+        destination: '',
+        startDate: today,
+        endDate: tomorrow,
+        rooms: 1,
+        adults: 2,
+        children: 0,
+        bundleSave: false,
+        addCar: false,
+        addFlight: false
+    });
+
+    const [isSticky, setIsSticky] = useState(false);
+
+    useEffect(() => {
+        const handleScroll = () => {
+            const offset = window.scrollY;
+            setIsSticky(offset > 500);
+        };
+
+        window.addEventListener('scroll', handleScroll);
+
+        return () => {
+            window.removeEventListener('scroll', handleScroll);
+        };
+    }, []);
+
+
+    const handleDateChange = (dates: [Date | null, Date | null]) => {
+        const [start, end] = dates;
+        setFormData(prevState => ({
+            ...prevState,
+            startDate: start || undefined,
+            endDate: end || undefined,
+        }));
+    };
+
+    const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const {name, value, type, checked} = e.target;
+        setFormData(prevState => ({
+            ...prevState,
+            [name]: type === 'checkbox' ? checked : value
+        }));
+    };
+
+    const handleHotelTypeChange = (type: 'single' | 'multi') => {
+        setFormData(prevState => ({
+            ...prevState,
+            hotelType: type
+        }));
+    };
+
+    const handleRoomDataChange = (key: 'rooms' | 'adults' | 'children', value: number) => {
+        setFormData(prevState => ({
+            ...prevState,
+            [key]: value
+        }));
+    };
+
+    const dispatch = useDispatch<AppDispatch>();
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        console.log('Form submitted with data:', formData);
+        dispatch(searchHotelList());
+    };
+
+    return (
+        <div className={`hotel-search-section ${isSticky ? 'sticky' : ''}`}>
+            <div className={`search-box ${isSticky ? 'sticky' : ''}`}>
+                {!isSticky &&
+                    <div>
+                        <h2>Searching for a place to stay?</h2>
+
+                        <div className="hotel-type-selection">
```

Comment:
This doesn't switch form based on hotel type. The multi type is supposed to have more fields